README.md

```diff
@@ -1,13 +1,191 @@
+>>> np.sin(0.0)
+0.0
+```
+
+However, the standard only specifies function inputs on `Array` objects. And
+indeed, some libraries, such as PyTorch, do not allow this:
+
+```py
+>>> import torch
+>>> torch.sin(0.0)
+Traceback (most recent call last):
+  File "<stdin>", line 1, in <module>
+TypeError: sin(): argument 'input' (position 1) must be Tensor, not float
+```
+
+In array-api-strict, this is also an error:
+
+```py
+>>> import array_api_strict as xp
+>>> xp.sin(0.0)
+Traceback (most recent call last):
+...
+AttributeError: 'float' object has no attribute 'dtype'
+```
+
+Here is an (incomplete) list of the sorts of ways that array-api-strict is
+strict/minimal:
+
+- Only those functions and methods that are [defined in the
+  standard](https://data-apis.org/array-api/latest/API_specification/index.html)
+  are included.
+
+- In those functions, only the keyword-arguments that are defined by the
+  standard are included. All signatures in array-api-strict use
+  [positional-only
+  arguments](https://data-apis.org/array-api/latest/API_specification/function_and_method_signatures.html#function-and-method-signatures).
+  As noted above, only array_api_strict array objects are accepted by
+  functions, except in the places where the standard allows Python scalars
+  (i.e., functions to not automatically call `asarray` on their inputs).
+
+- Only those [dtypes that are defined in the
+  standard](https://data-apis.org/array-api/latest/API_specification/data_types.html)
+  are included.
+
+- All functions and methods reject inputs if the standard does not *require*
+  the input dtype(s) to be supported. This is one of the most restrictive
+  aspects of the library. For example, in NumPy, most transcendental functions
+  like `sin` will accept integer array inputs, but the [standard only requires
+  them to accept floating-point
+  inputs](https://data-apis.org/array-api/latest/API_specification/generated/array_api.sin.html#array_api.sin),
+  so in array-api-strict, `sin(integer_array)` will raise an exception.
+
+- The
+  [indexing](https://data-apis.org/array-api/latest/API_specification/indexing.html)
+  semantics required by the standard are not
+
+- There are no distinct "scalar" objects as in NumPy. There are only 0-D
+  arrays.
+
+- Dtype objects are just empty objects that only implement [equality
+  comparison](https://data-apis.org/array-api/latest/API_specification/generated/array_api.data_types.__eq__.html).
+  The way to access dtype objects in the standard is by name, like
+  `xp.float32`.
+
+- The array object type itself is private and should not be accessed.
+  Subclassing or otherwise trying to directly initialize this object is not
+  supported. Arrays should created with one of the [array creation
+  functions](https://data-apis.org/array-api/latest/API_specification/creation_functions.html)
+  such as `asarray`.
+
+## Caveats
+
+array-api-strict is a thin pure Python wrapper around NumPy. NumPy 2.0 fully
```

Comment:
I suggest: "NumPy 2.0 has nearly full support for ....", since there are a few aspects that don't comply (e.g., returning scalars)

Reply:
I'll change this, but I was under the impression that scalars were not actually breaking compliance. As long as scalars duck-type as 0-D arrays, which they do, then it's just an implementation detail that numpy has a separate type for them.

Reply:
maybe, not sure if the duck-typing really is 100%. there's a few other reasons though, see the "not adopted part in the PR for NEP 56

Reply:
Where is NEP 56? Google comes up with nothing.

Reply:
https://github.com/numpy/numpy/pull/25542

Reply:
I'm going to continue this discussion over there since this is closed, but that NEP seems to just state that scalars aren't in compliance without actually explaining why. https://github.com/numpy/numpy/pull/25542#issuecomment-1909047579